qulice-findbugs/src/main/java/com/qulice/findbugs/FindBugsExcludes.java

```diff
@@ -0,0 +1,199 @@
+        return document;
+    }
+
+    /**
+     * Converts excludes to command argument
+     * as a path to the file.
+     * @return List of arguments (zero or one)
+     */
+    public final Collection<String> asArguments() {
+        final Document document = this.asXml();
+        final Element root = document.getRootElement();
+        final List<String> arguments = new LinkedList<>();
+        if (root.hasContent()) {
+            final File file = this.createTempFile();
+            FindBugsExcludes.writeXml(
+                file,
+                document
+            );
+            arguments.add(file.getAbsolutePath());
+        }
+        return arguments;
+    }
+
+    /**
+     * Writes xml to file.
+     * @param file File the XML will be written to.
+     * @param document The XML document that needs to be saved.
+     */
+    private static void writeXml(final File file, final Document document) {
+        try (final Writer writer = new OutputStreamWriter(
+            new FileOutputStream(file),
+            "UTF-8"
+        )) {
+            document.write(writer);
+        } catch (final IOException exc) {
+            throw new IllegalStateException(
+                "can't generate exclude rules for findbugs",
+                exc
+            );
+        }
+    }
+
+    /**
+     * Creates a temporary file.
+     * @return Temp file
+     */
+    private File createTempFile() {
+        return new File(
+            StringUtils.join(
+                this.env.tempdir().getPath(),
+                System.getProperty("path.separator"),
+                "findbug_excludes_",
+                String.valueOf(System.nanoTime()),
+                ".xml"
+            )
+        );
+    }
+
+    /**
+     * Converts the exclusion expressions to the FindBugs exclusion filter XML.
+     * http://findbugs.sourceforge.net/manual/filter.html
+     */
+    static class ExcludesXml {
+
+        /**
+         * Element names.
+         */
+        private static final String[] ELEMENTS = {
+            "Class", "Method", "Bug",
+        };
+
+        /**
+         * Argument names.
+         */
+        @SuppressWarnings("PMD.AvoidDuplicateLiterals")
+        private static final String[] ARGUMENTS = {
+            "name", "name", "pattern",
+        };
+
+        /**
+         * Root XML Element of the FindBugs exclusion filter.
+         */
+        private final Element root;
+
+        /**
+         * Constructor.
+         * @param root Root element
+         */
+        ExcludesXml(final Element root) {
+            this.root = root;
+        }
+
+        /**
+         * Converts the exclusion expression into a XML element.
+         * @param exclude An exclusion expression
+         */
+        public void addExclude(final String exclude) {
+            if (exclude != null && !exclude.isEmpty()) {
+                final Element match = new BaseElement("Match");
```

Comment:
@ilyakharlamov I think that using `cactoos` you can make this method less procedural, WDYT?

Reply:
@paulodamaso fixed